server.go

```diff
@@ -88,6 +92,10 @@ func (app *App) securityCredentialsHandler(w http.ResponseWriter, r *http.Reques
 	write(w, app.RoleName)
 }
 
+func (app *App) trailingSlashRedirect(w http.ResponseWriter, r *http.Request) {
```

Comment:
I don't think the ec2-metadata API does that (but I could remember wrong). If it doesn't then we should not either because we should mock its behaviour as close as possible.